src/main/java/com/gtnewhorizons/modularui/api/screen/ModularUIContext.java

```diff
@@ -271,6 +272,10 @@ public void closeAllButMain() {
     public void storeWindowPos(ModularWindow window, Pos2d pos) {
         if (windows.contains(window)) {
             this.lastWindowPos.put(window, pos);
```

Comment:
I think you can skip `lastWindowPos` if sync id is not null, as it probably won't be referenced again.
```
if (id != null) {
    // put synced pos
} else {
    // put non-synced pos
}
```